mono/mini/mini.c

```diff
@@ -4125,6 +4126,12 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 
 	mono_domain_lock (target_domain);
 
+	if (mono_stats_method_desc && mono_method_desc_full_match (mono_stats_method_desc, method)) {
+		g_printf ("Printing JIT stats at method: %s\n", mono_method_get_full_name (method));
+		mono_print_jit_stats ();
+		g_printf ("\n");
```

Comment:
This should not be necessary. If you need an extra newline, it goes inside the stats printing function.